bin/index.js

```diff
@@ -10,22 +10,55 @@ const {
   generatePdfFromBuildWithConfig,
 } = require("../lib");
 
+function generatePdfOptions({ sandbox, margin, format, printBackground }) {
+  return {
+    puppeteerArgs: sandbox ? [] : ["--no-sandbox"],
+    puppeteerPdfOpts: { margin, format, printBackground },
+  };
+}
+
+function parseMargin(marginString) {
+  const matches = [...marginString.match(/\d+\w{0,2}/g)];
+  if (matches.length !== 4) {
+    throw Error(
+      `Was expecting exactly 4 margin specifiers, instead got ${matches.length}. Margin specifier was "${marginString}".`
+    );
+  }
+  const [top, right, bottom, left] = matches;
+  return { top, right, bottom, left };
+}
+
 program
   .version(require("../package.json").version)
   .name("docusaurus-pdf")
   .usage("<initialDocsUrl> [filename]")
-  .description("Generate PDF from initial docs url")
-  .arguments("<initialDocsUrl> [filename]")
-  .action((initialDocsUrl, filename) => {
-    generatePdf(initialDocsUrl, filename)
-      .then(() => {
-        console.log(chalk.green("Finish generating PDF!"));
-        process.exit(0);
-      })
-      .catch((err) => {
-        console.error(chalk.red(err.stack));
-        process.exit(1);
-      });
+  .description("Generate a PDF from a docusaurus website")
+  .option("--no-sandbox", "Start puppeteer with --no-sandbox flag")
+  .option(
+    "--margin <margin>",
+    "Set margins of the pdf document with units in order top, right, bottom, left (units px,in,mm,cm)",
+    parseMargin,
+    "25px 35px 25px 35px"
+  )
+  .option(
+    "--format <format>",
+    "Set the size of the page, e.g. (A4, Letter)",
+    "A4"
+  )
+  .option("--no-print-background", "Disable printing the page background");
+
+program
+  .command("from-website <initialDocsUrl> [filename]", {
+    isDefault: true,
+  })
+  .description("Generate PDF from an already hosted website")
+  .action(async (initialDocsUrl, filename = "docusaurus.pdf") => {
+    await generatePdf(
+      initialDocsUrl,
+      filename,
+      generatePdfOptions(program.opts())
+    );
+    console.log(chalk.green("Finish generating PDF!"));
   });
 
 program
```

Comment:
Is there a way to show the new options `--margin` and `--format` for the other commands as well?
If I run one of the following I don't see those options listed:

```
node bin/index.js from-website --help
node bin/index.js from-build --help
node bin/index.js from-build-config --help
```

I think it would be helpful to see the options on every command which supports them.

Reply:
I think they currently only show up when you run `node bin/index.js --help`, since they're global options for all commands.

I'll see if I can add a message saying something like:

```sh
$ node bin/index.js from-website --help
Usage: docusaurus-pdf from-website [options] <initialDocsUrl> [filename]

Generate PDF from an already hosted website

Global options can be seen by running `docusaurus-pdf --help`

Options:
  -h, --help  display help for command
```